tests/unit/modules/test_x509.py

```diff
@@ -186,6 +187,360 @@ def test_create_certificate(self):
+                                          Email='certadm@example.org',
+                                          basicConstraints="critical CA:true",
+                                          keyUsage="critical cRLSign, keyCertSign",
+                                          subjectKeyIdentifier='hash',
+                                          authorityKeyIdentifier='keyid,issuer:always',
+                                          days_valid=3650,
+                                          days_remaining=0)
+
+        not_after_str = 'this is an intentionally wrong format'
+
+        # Try to sign a new server certificate with the wrong date
+        try:
+            x509.create_certificate(text=True,
+                                    signing_private_key=ca_key,
+                                    signing_cert=ca_cert,
+                                    CN='Redacted Normal Certificate',
+                                    O='Redacted',
+                                    C='BE',
+                                    ST='Antwerp',
+                                    L='Local Town',
+                                    Email='certadm@example.org',
+                                    basicConstraints="critical CA:false",
+                                    keyUsage="critical keyEncipherment",
+                                    subjectKeyIdentifier='hash',
+                                    authorityKeyIdentifier='keyid,issuer:always',
+                                    not_after=not_after_str,
+                                    days_remaining=0)
+            self.fail('Certificate generation must fail as not_after is wrong')
+        except Exception as e:
+            error = str(e)
+
+        msg = "not_after: u'this is an intentionally wrong format' is not in required format %Y-%m-%d %H:%M:%S"
+        self.assertEqual(msg, e.message)
+
+    @skipIf(not HAS_M2CRYPTO, 'Skipping, M2Crypto is unavailble')
+    def test_create_certificate_with_not_before_and_not_after(self):
+        ca_key = '''
+-----BEGIN RSA PRIVATE KEY-----
+MIICWwIBAAKBgQCjdjbgL4kQ8Lu73xeRRM1q3C3K3ptfCLpyfw38LRnymxaoJ6ls
+pNSx2dU1uJ89YKFlYLo1QcEk4rJ2fdIjarV0kuNCY3rC8jYUp9BpAU5Z6p9HKeT1
+2rTPH81JyjbQDR5PyfCyzYOQtpwpB4zIUUK/Go7tTm409xGKbbUFugJNgQIDAQAB
+AoGAF24we34U1ZrMLifSRv5nu3OIFNZHyx2DLDpOFOGaII5edwgIXwxZeIzS5Ppr
+yO568/8jcdLVDqZ4EkgCwRTgoXRq3a1GLHGFmBdDNvWjSTTMLoozuM0t2zjRmIsH
+hUd7tnai9Lf1Bp5HlBEhBU2gZWk+SXqLvxXe74/+BDAj7gECQQDRw1OPsrgTvs3R
+3MNwX6W8+iBYMTGjn6f/6rvEzUs/k6rwJluV7n8ISNUIAxoPy5g5vEYK6Ln/Ttc7
+u0K1KNlRAkEAx34qcxjuswavL3biNGE+8LpDJnJx1jaNWoH+ObuzYCCVMusdT2gy
+kKuq9ytTDgXd2qwZpIDNmscvReFy10glMQJAXebMz3U4Bk7SIHJtYy7OKQzn0dMj
+35WnRV81c2Jbnzhhu2PQeAvt/i1sgEuzLQL9QEtSJ6wLJ4mJvImV0TdaIQJAAYyk
+TcKK0A8kOy0kMp3yvDHmJZ1L7wr7bBGIZPBlQ0Ddh8i1sJExm1gJ+uN2QKyg/XrK
+tDFf52zWnCdVGgDwcQJALW/WcbSEK+JVV6KDJYpwCzWpKIKpBI0F6fdCr1G7Xcwj
+c9bcgp7D7xD+TxWWNj4CSXEccJgGr91StV+gFg4ARQ==
+-----END RSA PRIVATE KEY-----
+'''
```

Comment:
Same goes for this `ca_key` - if it's the same key used in other places you should extract it.

Reply:
Externalization of default args also done here!